### ISSUE TITLE
Bug on Sales return discount Value

### ISSUE THREAD

**Issue by MufassalAmeer** [opensourcepos/opensourcepos#3097]
### Background information

__IMPORTANT: If you choose to ignore this issue report template, your issue will be closed as we cannot help without the requested information.__

Please make sure you tick (add an x between the square brackets with no spaces) the following check boxes:

- [] Reporting an issue of an unmodified OSPOS installation
- [] Checked [open and closed issues](https://github.com/opensourcepos/opensourcepos/issues?utf8=%E2%9C%93&q=is%3Aissue) and no similar issue was already reported (please make sure you searched!)
- [] Read [README](https://github.com/opensourcepos/opensourcepos/blob/master/README.md), [WHATS_NEW](https://github.com/opensourcepos/opensourcepos/blob/master/WHATS_NEW.txt), [INSTALL.md](https://github.com/opensourcepos/opensourcepos/blob/master/INSTALL.md) and [UPGRADE](https://github.com/opensourcepos/opensourcepos/blob/master/UPGRADE.txt)
- [] Read the [FAQ](https://github.com/opensourcepos/opensourcepos#faq) for any known install and/or upgrade gotchas (in specific PHP extensions installed)
- [] Read the [wiki](https://github.com/opensourcepos/opensourcepos/wiki)
- [] Executed any database upgrade scripts if an upgrade pre 3.0.0 (e.g. database/2.4_to_3.0.sql)
- [] Aware the installation code is in [bintray](https://bintray.com/jekkos/opensourcepos/opensourcepos/view/files?sort=updated&order=asc#files) (see README), and [GitHub master](https://github.com/opensourcepos/opensourcepos/tree/master) is for [developers only](https://github.com/opensourcepos/opensourcepos/wiki/Development-setup) and therefore not complete nor stable

### Installation information

- OSPOS version is: 3.3.3 - 
- OSPOS git commit hash is: c5b22a
- PHP version is:  7.3.26
- MySQL or MariaDB version is: 5.5.5-10.4.17-MariaDB
- OS and version is: Windows NT 10.0
- WebServer is: Apache/2.4.46 (Win64) OpenSSL/1.1.1h PHP/7.3.26
- Selected language is:  en-US
- (If applicable) Docker installation: 
- (If applicable) Installation package for the LAMP/LEMP stack is: 

### Issue / Bug  / Question / New Feature

Please write your issue here. If a bug, please make sure to provide as much information as possible including configuration settings (e.g. Decimals set, Tax mode), language and steps to reproduce the bug.

Hi Team,

When we are returning a sale with a discount value not percentage. the amount is not dedcuted. insted it adds it to the total.
i have attached a screenshot for refrence. 
![image](https://user-images.githubusercontent.com/26744995/107624659-39215980-6c81-11eb-8139-6ba76251e2a9.png)
Recept 
Actally Charged amount - 1100
![image](https://user-images.githubusercontent.com/26744995/107624797-640bad80-6c81-11eb-8692-83d45267d868.png)
if try to return the same recipt 
it shows we need to return - 1300
syatem has added up the discount to return cash

**Comment by odiea** [comment]
The discount works fine on a return if it is a % value but if it is a $ value It adds the value to the Return total. Clear as mud.

**Comment by odiea** [comment]
I have tried the following and it appears to fix the issue. application/libraries/Sale_lib and add the following to the public function return_entire_sale($receipt_sale_id) beneath the foreeach and see if it fixes this issue. Thanks
```
if($row->discount_type == 1)
	{
		$row->discount = -$row->discount; 			
	}
```

![2021-02-12 10_43_36-Window](https://user-images.githubusercontent.com/10962177/107803442-cdda9480-6d1f-11eb-8d19-9672c9294e0d.png)

**Comment by SteveIreland** [comment]
I believe I included a fix for  this in #3107.   I think the main issue was that it was missing the code that computes the discount amount when the discount is a fixed amount.  I added it back and my brief tests showed that the discount is now being computed correctly.

**Comment by MufassalAmeer** [comment]
Thank you Very Much. Working Fine now. Thank you Once again. 
![image](https://user-images.githubusercontent.com/26744995/107869111-a535d480-6eb0-11eb-8e10-c82d15ce99ca.png)
